.github/workflows/unit-tests.yml

```diff
@@ -340,21 +344,21 @@ jobs:
       - name: Set up Python Dev Version
         uses: actions/setup-python@v4
         with:
-          python-version: '3.11-dev'
+          python-version: '3.12-dev'
 
       - name: Install dependencies
         run: |
           python --version
-          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade pip setuptools wheel meson[ninja]==1.0.1 meson-python==0.13.1
```

Comment:
Based on https://github.com/pandas-dev/pandas/pull/54244 we don't need the `[ninja]`?

Reply:
I'd prefer not to rely on the Github runners having ninja installed.

Reply:
As mentioned in the linked ticket -- you only don't need it in the sense that pip install will add ninja automatically as an additional build-requires, comparable to if it were listed in pyproject.toml

In this case, you are telling `pip install -e .` to avoid build isolation, and it turns out that pip will therefore not install build-requires as needed. This is why all the relevant build deps are installed in their own separate step, manually, without using meson-python to resolve them.

That means you either need to unconditionally install ninja, or you need to check whether it is good enough according to meson-python's internal criteria... and it's easiest to simply install it.